### PR TITLE
add osx_arm64 build

### PR DESCRIPTION
**Pull request by matthiasdiener**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
FYI - I was able to compile this locally (natively) after applying the following patch:
```diff
diff --git a/recipe/build.sh b/recipe/build.sh
index c8ca87c..dc8529b 100755
--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -23,6 +23,7 @@ if [[ "$PKG_NAME" == "scotch" ]]; then
   # build
   cd src/
   make esmumps
+  cd ..
 
   # install
   mkdir -p $PREFIX/lib/
```

I uploaded the resulting packages to https://anaconda.org/tobiasrobotics/

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin please restart ci

**Comment by Tobias-Fischer** [comment]
@matthiasdiener any chance you can give me write access to your fork?

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin please rerender

**Comment by matthiasdiener** [comment]
> @matthiasdiener any chance you can give me write access to your fork?

Sure, you should have gotten an invitation.

**Comment by Tobias-Fischer** [comment]
So it seems like this will need considerable work to get to cross-compile unfortunately.

They use an executable that needs to run on the build platform to generate some header files (see https://gitlab.inria.fr/scotch/scotch/-/blob/master/src/libscotch/CMakeLists.txt#L52-111). So a native build for osx-64 has to be done first, and then the cross-compile build. General how-to if anyone is keen on giving it a go: https://cmake.org/cmake/help/book/mastering-cmake/chapter/Cross%20Compiling%20With%20CMake.html#running-executables-built-in-the-project

**Comment by Tobias-Fischer** [comment]
Also I'm not sure why we now see a linker error in the osx-64 build:
```
x86_64-apple-darwin13.4.0-clang -shared -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/runner/miniforge3/conda-bld/scotch-pkgs_1645054245672/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib -L/Users/runner/miniforge3/conda-bld/scotch-pkgs_1645054245672/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib -lm -lz -pthread -lm -lz -pthread -Wl,-install_name,@rpath/libptscotcherr-6.dylib library_error_pt.o -o libptscotcherr-6.dylib
ld: warning: -pie being ignored. It is only used when linking a main executable
Undefined symbols for architecture x86_64:
  "_MPI_Comm_rank", referenced from:
      _SCOTCH_errorPrint in library_error_pt.o
      _SCOTCH_errorPrintW in library_error_pt.o
  "_MPI_Initialized", referenced from:
      _SCOTCH_errorPrint in library_error_pt.o
      _SCOTCH_errorPrintW in library_error_pt.o
  "_ompi_mpi_comm_world", referenced from:
      _SCOTCH_errorPrint in library_error_pt.o
      _SCOTCH_errorPrintW in library_error_pt.o
ld: symbol(s) not found for architecture x86_64
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:3120: libptscotcherr.a] Error 1
```

**Comment by matthiasdiener** [comment]
> So it seems like this will need considerable work to get to cross-compile unfortunately.
> 
> They use an executable that needs to run on the build platform to generate some header files (see https://gitlab.inria.fr/scotch/scotch/-/blob/master/src/libscotch/CMakeLists.txt#L52-111). So a native build for osx-64 has to be done first, and then the cross-compile build. General how-to if anyone is keen on giving it a go: https://cmake.org/cmake/help/book/mastering-cmake/chapter/Cross%20Compiling%20With%20CMake.html#running-executables-built-in-the-project

Maybe it is enough to compile dummysizes/ptdummysizes with the host compiler.

**Comment by Tobias-Fischer** [comment]
It probably is, but I'm not CMake guru enough to work that out .. @traversaro - sorry to ping you - do you have any ideas what the easiest way forward would be?

**Comment by traversaro** [comment]
> It probably is, but I'm not CMake guru enough to work that out .. @traversaro - sorry to ping you - do you have any ideas what the easiest way forward would be?

Sorry, I had lost this mention! Fortunatly I considered "check what's the status of recursive ipopt transitive dependency on osx-64" a fun activity anyhow. :D

**Comment by traversaro** [comment]
So, a bit of the things that I understood:

## CMake build system not available 
* The CMake build system for scotch is available since scotch 7, while we are building scotch 6.0.9 at the moment, so the only build system available is the Makefile one. If we ever switch to CMake, we need to remember to to this at the same time of a version update to avoid ABI issues (fortunatly scotch's run_exports has `max_pin='x.x.x'` https://github.com/conda-forge/scotch-feedstock/blob/013a226cdd48dd3919084f3321e5628740062578/recipe/meta.yaml#L28, so even a patch version update will be enough)

## dummysize and ptdummysize
I inspected a bit what the `dummysize` tool is doing, and apparently is computing a bunch of paddings/scrut sizes via sizeof() (see https://gitlab.inria.fr/scotch/scotch/-/blob/v6.0.9/src/libscotch/dummysizes.c#L240). That means that running `dummysize` on the build platform (i.e. osx intel64) may not work and introduce subtle bugs, because nothing ensures that a `sizeof` of a structure on osx intel64 is the same of osx arm64. For cross-compilation to work, a osx-arm64 dummysize (either manually, or via emulation that I do not think is available) should produce the headers:
  * `scotch.h`
  * `scotchf.h`
  * `ptscotch.h`
  * `ptscotchf.h`

Then, just for cross-compilation builds we should comment out the rules to build those files from the make file (i.e. from mhttps://gitlab.inria.fr/scotch/scotch/-/blob/v6.0.9/src/libscotch/Makefile#L3079 to https://gitlab.inria.fr/scotch/scotch/-/blob/v6.0.9/src/libscotch/Makefile#L3093) as those files are already available and should not be generated as part of the build. Probaly other modifications may be necessary to handle the installation of those files.

In a nutshell, we may take those headers from @Tobias-Fischer's builds, but we should add a logic to clearly fail as soon as the version of scotch is bumped, as in that case those headers should be generated again manually.

**Comment by Tobias-Fischer** [comment]
Thanks for that, @traversaro!

What do you think of https://github.com/conda-forge/scotch-feedstock/pull/61/commits/6fd52394c807f7dba819c79572c79a738080d23c ?

It's using my pregenerated headers. The headers have the version in the filename, so any bump will clearly fail. Seems like this is the only proper way - unfortunately it does require manually generating the headers on version bumps, but I'd be happy to do that.

**Comment by Tobias-Fischer** [comment]
Hi @matthiasdiener - do you know what's going on with the osx-64 builds? Seems like we need to sort these out ..

**Comment by Tobias-Fischer** [comment]
Clearly I don't have a clue what I'm doing.. if anyone has an idea how to get osx to build that would be amazing @conda-forge/help-osx-arm64 @matthiasdiener @traversaro - I'll stop playing around

**Comment by traversaro** [comment]
Probably I am stating the obvious, but looking at the osx-64 errors: for some reason the source files of the `libptscotcherr-6.dylib` library are compiled with the normal compilers, as opposed to `mpicc`  that apparently should be used (based on the `#ifdef` contained in the files contained in the library) instead and for example is used for `libptscotch-6.dylib`.

**Comment by traversaro** [comment]
By reading a bit better the logs, it seems that a version of `libptscotcherr-6.dylib` built with `mpicc` was already built, so for some reason the library is built again with the wrong "compiler", I will try to understand why.

**Comment by traversaro** [comment]
Apparently on the Linux side instead `libptscotcherr-6.so` is generated only once and only by mpicc, but it is generated two times if make is invoked with `--just-print`, even if in that case the execution fails. I am not an expert of debugging Makefiles, but I found https://www.oreilly.com/openbook/make3/book/ch12.pdf usefuk.

**Comment by traversaro** [comment]
> Apparently on the Linux side instead `libptscotcherr-6.so` is generated only once and only by mpicc, but it is generated two times if make is invoked with `--just-print`, even if in that case the execution fails. I am not an expert of debugging Makefiles, but I found https://www.oreilly.com/openbook/make3/book/ch12.pdf usefuk.

Running make with the option `-d` is providing some insight, apparently `libptscotcherr-6.so`` is created again because:
~~~
Prerequisite 'library_error_pt.o' is newer than target 'libptscotcherr.a'.
2022-03-07T08:16:18.0490650Z       Must remake target 'libptscotcherr.a'.
~~~
The problem is that is happening in a Makefile for which the wrong environment variables are not set, so the compilation fails. Not sure why `library_error_pt.o` results being newer.

**Comment by traversaro** [comment]
I guess for some reason the `library_error_pt.o` and the `libptscotcherr.a` files are created/modified close in time, and for some reason the OS or `make` believes that  `libptscotcherr.a` is created/modified before `library_error_pt.o`. As a workaround, I added a small sleep before the creation of `libptscotcherr.a`, to ensure that there is no confusion (see https://github.com/conda-forge/scotch-feedstock/pull/65/commits/59cf7926278c75086176f9a516575fe6d9a5460f). With that workaround, now only `osx_arm64_mpiopenmpi` job is failing, but for a different reason.

**Comment by traversaro** [comment]
The `osx_arm64_mpiopenmpi ` job is failing with:
~~~
2022-03-07T11:23:36.7107380Z make[2]: Nothing to be done for 'ptscotch.h'.
2022-03-07T11:23:36.7108820Z make[2]: Nothing to be done for 'ptscotchf.h'.
2022-03-07T11:23:36.7115720Z mpicc -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/scotch-pkgs_1646651825961/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646651825961/work=/usr/local/src/conda/ptscotch-6.0.9 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646651825961/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho=/usr/local/src/conda-prefix -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646651825961/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646651825961/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -DSCOTCH_PTSCOTCH -fPIC -c bdgraph.c -o bdgraph.o -DSCOTCH_VERSION_NUM=6 -DSCOTCH_RELEASE_NUM=0 -DSCOTCH_PATCHLEVEL_NUM=9
2022-03-07T11:23:36.7121030Z make[2]: mpicc: Bad CPU type in executable
2022-03-07T11:23:36.7122110Z make[2]: *** [Makefile:50: bdgraph.o] Error 127
2022-03-07T11:23:36.7124140Z make[2]: Leaving directory '/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646651825961/work/src/libscotch'
2022-03-07T11:23:36.7124960Z make[1]: *** [Makefile:72: ptscotch] Error 2
2022-03-07T11:23:36.7126160Z make[1]: Leaving directory '/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646651825961/work/src/libscotch'
2022-03-07T11:23:36.7127070Z make: *** [Makefile:103: libptscotch] Error 2
~~~

However, the same command in `osx_arm64_mpimpich`, as if in that case `mpicc` is pointing to the correct cross-compiled. Anyone has a clue?

**Comment by traversaro** [comment]
> However, the same command in `osx_arm64_mpimpich`, as if in that case `mpicc` is pointing to the correct cross-compiled. Anyone has a clue?

Based on https://github.com/conda-forge/libnetcdf-feedstock/pull/120#issuecomment-822064958 I tried https://github.com/conda-forge/scotch-feedstock/pull/65/commits/e2c0a4b5ecf50ee55e0a653532e7770fc2c63031, let's see if it does something.

**Comment by traversaro** [comment]
Ok, I finally got it to work.  @Tobias-Fischer 

The missing fixes were:

## Sleep Workaround

Add a sleep 1 to avoid that the `libptscotcherr.a` needs to be re-built as `library_error_pt.o` appears to be newer: https://github.com/conda-forge/scotch-feedstock/pull/66/commits/b9ff51075eb30f041f3fb93d33ff961d3d10b8bd . This is a pure workaround, but as we can remove it as soon as we stitch to the CMake build system I think it is ok to have it

## Fix cross-compilation with mpicc when using openmpi:
Commits: https://github.com/conda-forge/scotch-feedstock/pull/66/commits/f4d7a6995ceb49103b1d2b7dcb45791df57c0edf and https://github.com/conda-forge/scotch-feedstock/pull/66/commits/e6baf511af1e42a88115aac155d54b7f0bd4d4a2 . 

Cross-compilation with `mpich` works out of the box as:
  * `mpicc` from `mpich` is just a bash script, so it works fine also the one installed in the host prefix
  * `mpicc` from `mpich` observes the `CC` env variable, so it uses automatically the cross-compiled

However, this is not the case for `openmpi`, in particular:
  * `mpicc` from `openmpi` is a compiled binary, so to execute it it is necessary to install it also in the build prefix (see https://github.com/conda-forge/scotch-feedstock/commit/f4d7a6995ceb49103b1d2b7dcb45791df57c0edf)
  * `mpicc` from `openmpi` has its own custom env variable `OMPI_CC`, so it is necessary to define it properly to make sure that `mpicc` works fine when cross-compiling (see https://github.com/conda-forge/scotch-feedstock/commit/e6baf511af1e42a88115aac155d54b7f0bd4d4a2)
  
  
With these changes, all the jobs works fine. Feel free to merge one of the following two PRs:
* https://github.com/matthiasdiener/scotch-feedstock/pull/2 updadates this PR
* https://github.com/conda-forge/scotch-feedstock/pull/66 is a PR that supersedes this one (useful to check CI)

**Comment by traversaro** [comment]
The last open point is the point of hardcoded header in https://github.com/conda-forge/scotch-feedstock/pull/66#discussion_r820989566 from @isuruf .

**Comment by traversaro** [comment]
>  Can you check if the two headers match for arm64 vs intel?

If I am not wrong (I did this manually so I may have created an error) `scotch.h` are identical, while for `ptscotch.h` the difference is:
~~~diff
--- <unnamed>
+++ <unnamed>
@@ -95,7 +95,7 @@
 proper padding                               +*/
 
 typedef struct {
-  double                    dummy[29];
+  double                    dummy[30];
 } SCOTCH_Dgraph;
 
 typedef struct {
@@ -107,7 +107,7 @@
 } SCOTCH_Dmapping;
 
 typedef struct {
-  double                    dummy[13];
+  double                    dummy[14];
 } SCOTCH_Dordering;
 
 /*
~~~

**Comment by Tobias-Fischer** [comment]
Hi @traversaro - great debugging work, many thanks to pushing this PR forward! Looks like we're getting very close now :).

I'd be happy to create the header files for new versions.

**Comment by isuruf** [comment]
Let's try to figure out why those two are differing.

**Comment by Tobias-Fischer** [comment]
I just did another manual comparison of the header files. For me there is no difference between the files. If that's the case, we can indeed go ahead and build a native version, and use these header files for the cross-compiled build. @traversaro - how did you go about comparing the header files?

My approach was to:
- on a osx-64 machine, build scotch with `conda mambabuild . -m ../.ci_support/osx_64_mpiopenmpi.yaml`
- on a osx-arm64 machine, build scotch with `conda mambabuild . -m ../.ci_support/osx_arm64_mpiopenmpi.yaml` after removing the `use_predefined_headers.patch` and `use_predefined_headers_metis.patch` patches as well as removing the  section in the build.sh that currently copies the header files.

**Comment by traversaro** [comment]
> how did you go about comparing the header files?

How compared the comitted header with one that I got from an archive, but now I noticed that the size of the `SCOTCH_Dordering` and `SCOTCH_Dgraph` strcutres actually changes depending on wheter scotch is built with `openmpi` or with `mpich`, so probably I was comparing two different headers. This means that hardcoding the headers, if we are not accounting for this is not only a mantainance burden, but actually wrong! So it probably make sense to just swicht to use `dummysize` and `ptdummysize` from the build as @isuruf suggested.

**Comment by ngam** [comment]
> Ok, I finally got it to work. @Tobias-Fischer
> 
> The missing fixes were:
> 
> ## Sleep Workaround
> Add a sleep 1 to avoid that the `libptscotcherr.a` needs to be re-built as `library_error_pt.o` appears to be newer: [b9ff510](https://github.com/conda-forge/scotch-feedstock/commit/b9ff51075eb30f041f3fb93d33ff961d3d10b8bd) . This is a dirty workaround that I am not particurlarly proud of, but as we can remove it as soon as we switch to the CMake build system I think it is ok to have it there.
> 
> ## Fix cross-compilation with mpicc when using openmpi
> Commits: [f4d7a69](https://github.com/conda-forge/scotch-feedstock/commit/f4d7a6995ceb49103b1d2b7dcb45791df57c0edf) and [e6baf51](https://github.com/conda-forge/scotch-feedstock/commit/e6baf511af1e42a88115aac155d54b7f0bd4d4a2) .
> 
> Cross-compilation with `mpich` works out of the box as:
> 
> * `mpicc` from `mpich` is just a bash script, so it works fine also the one installed in the host prefix
> * `mpicc` from `mpich` observes the `CC` env variable, so it uses automatically the cross-compiled
> 
> However, this is not the case for `openmpi`, in particular:
> 
> * `mpicc` from `openmpi` is a compiled binary, so to execute it it is necessary to install it also in the build prefix (see [f4d7a69](https://github.com/conda-forge/scotch-feedstock/commit/f4d7a6995ceb49103b1d2b7dcb45791df57c0edf))
> * `mpicc` from `openmpi` has its own custom env variable `OMPI_CC`, so it is necessary to define it properly to make sure that `mpicc` works fine when cross-compiling (see [e6baf51](https://github.com/conda-forge/scotch-feedstock/commit/e6baf511af1e42a88115aac155d54b7f0bd4d4a2))
> 
> With these changes, all the jobs works fine. Feel free to merge one of the following two PRs:
> 
> * [Add fixes to cross-compile on osx arm64 matthiasdiener/scotch-feedstock#2](https://github.com/matthiasdiener/scotch-feedstock/pull/2) updadates this PR
> * [Fix cross-compilation on osx arm64 #66](https://github.com/conda-forge/scotch-feedstock/pull/66) is a PR that supersedes this one (useful to check CI)

Looks like this is almost fixed, but in case more time is needed, would it be reasonable to simply skip the openmpi build for now to enable the mpich/nompi to go through? We could then take our time fixing the remaining issues.

**Comment by traversaro** [comment]
I switched to use `CC_FOR_BUILD` in https://github.com/conda-forge/scotch-feedstock/pull/67 . Unfortunatly I had to patch the makefiles as the recursive use of make effectively did not permit to use the `CCD` env variable that should be used to specify the compiler used to compiled `ptdummysize`.

**Comment by traversaro** [comment]
> Looks like this is almost fixed, but in case more time is needed, would it be reasonable to simply skip the openmpi build for now to enable the mpich/nompi to go through? We could then take our time fixing the remaining issues.

All the openmpi-specific issue were solved.

**Comment by Tobias-Fischer** [comment]
Awesome work Silvio! I think this is ready for review/merge :)

**Comment by matthiasdiener** [comment]
Thank you for the great work @traversaro @Tobias-Fischer ! This is ready for review.

File: recipe/build.sh [hunk 1]
```diff
@@ -11,49 +15,51 @@ if [[ $(uname) == "Darwin" ]]; then
...
+
+  export CCP=mpicc
+  # This is only used by open-mpi's mpicc
+  # ignored in other cases
+  export OMPI_CC=$CC
+  export OPAL_PREFIX=$PREFIX
+
+  # build
+  cd src/
+  make ptesmumps
```
Comment (isuruf) [inline recipe/build.sh:51]: If you do, then you don't need the new patch
```suggestion
  make ptesmumps CCD=$CC_FOR_BUILD
```

**Comment by basnijholt** [comment]
Wow, great work!

Everything looks good to me, especially since the tests are passing.

**Review by basnijholt — APPROVED** [review]

File: recipe/build.sh (hunk 1 above)
Comment (matthiasdiener) [reply recipe/build.sh:51]: If I understood https://github.com/conda-forge/scotch-feedstock/pull/67#issuecomment-1061875549 correctly, setting CCD might not be enough?

File: recipe/build.sh [hunk 2]
```diff
@@ -11,49 +15,51 @@ if [[ $(uname) == "Darwin" ]]; then
...
-cp -v include/ptscotch*.h $PREFIX/include/
-# avoid conflicts with the real parmetis.h
-mkdir -p $PREFIX/include/scotch
-cp -v include/parmetis.h  $PREFIX/include/scotch/
+
+if [[ "$PKG_NAME" == "scotch" ]]; then
+
+  # build
+  cd src/
+  make esmumps
```
Comment (isuruf) [inline recipe/build.sh:27]:
```suggestion
  make esmumps CCD=$CC_FOR_BUILD
```

File: recipe/build.sh (hunk 1 above)
Comment (traversaro) [reply recipe/build.sh]: I may be wrong (I am definititely not a Make expert) but in the end I went for the patch as it seemed to me that in https://gitlab.inria.fr/scotch/scotch/-/blob/v6.0.9/src/libscotch/Makefile#L64 there was some recursive make logic that was basically overwriting the `CCD` value.

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin please restart ci

File: recipe/build.sh (hunk 1 above)
Comment (Tobias-Fischer) [reply recipe/build.sh:51]: For future reference, this fails: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=470850&view=logs&jobId=7afb22dc-3239-5456-794e-5378c2396929

```
+ make esmumps CCD=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_build_env/bin/x86_64-apple-darwin13.4.0-clang
(cd libscotch ;      make VERSION=6 RELEASE=0 PATCHLEVEL=9 scotch && make install)
make[1]: Entering directory '/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/work/src/libscotch'
make 					\
				CC="arm64-apple-darwin20.0.0-clang"					\
				CCD="arm64-apple-darwin20.0.0-clang"					\
				scotch.h					\
				scotchf.h					\
				libscotch.a					\
				libscotcherr.a				\
				libscotcherrexit.a
make[2]: Entering directory '/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/work/src/libscotch'
arm64-apple-darwin20.0.0-clang -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/work=/usr/local/src/conda/scotch-6.0.9 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho=/usr/local/src/conda-prefix -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -fPIC -c arch.c -o arch.o -DSCOTCH_VERSION_NUM=6 -DSCOTCH_RELEASE_NUM=0 -DSCOTCH_PATCHLEVEL_NUM=9
arm64-apple-darwin20.0.0-clang -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/work=/usr/local/src/conda/scotch-6.0.9 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho=/usr/local/src/conda-prefix -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -O3 -I/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/include -DIDXSIZE64 -DSCOTCH_RENAME -Drestrict=__restrict -DCOMMON_FILE_COMPRESS_GZ -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_PTHREAD -DSCOTCH_PTHREAD -DCOMMON_PTHREAD_BARRIER -DCOMMON_TIMING_OLD -DSCOTCH_VERSION_NUM=6 -DSCOTCH_RELEASE_NUM=0 -DSCOTCH_PATCHLEVEL_NUM=9 dummysizes.c -o dummysizes -Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib -L/Users/runner/miniforge3/conda-bld/scotch-pkgs_1646774941871/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib -lm -lz -pthread -lm -lz -pthread -lm -lz -pthread
./dummysizes "-s" library.h scotch.h
/bin/sh: ./dummysizes: Bad CPU type in executable
```

**Comment by jschueller** [comment]
thanks for all the work guys!

File: recipe/build.sh (hunk 1 above)
Comment (traversaro) [reply recipe/build.sh:51]: Perhaps a simpler patch could be to remove the line https://gitlab.inria.fr/scotch/scotch/-/blob/v6.0.9/src/libscotch/Makefile#L64, but to be honest I am not 100% sure of the other side effects of that patch, while the existing patch of using `CC_FOR_BUILD` only when I know it was necessary it was more clear to me.

**Review by minrk — APPROVED** [review]

**Comment by dalcinl** [comment]
@matthiasdiener Should I merge? Please confirm.

**Review by dalcinl — APPROVED** [review]
